node/src/components/chain_synchronizer/operations.rs

```diff
@@ -1896,7 +1878,7 @@ where
             .await?
         {
             Some(ctx) => ctx,
-            None => return Ok(FastSyncOutcome::ShouldCommitGenesis),
+            None => return Err(Error::NoBlocksInStorage),
```

Comment:
Might be slightly simpler changing `ChainSyncContext::new_for_fast_sync` to return `Result<ChainSyncContext<'a, REv>, Error>`

Also, the comment just below here starting with "We should have at least one block header in storage now..." should probably be deleted.

Reply:
I'll leave the comment because when I change the chainspec loader to not create the immediate switch block at genesis if the current timestamp is after genesis, it might still be the case that we'll start `run_fast_sync_task` without any blocks in storage, and only `ChainSyncContext::new_for_fast_sync` will download the one at trusted hash.